doc/rados/operations/crush-map-edits.rst

```diff
@@ -240,121 +251,137 @@ and two node buckets named ``host`` and ``rack`` respectively.
-	   non-uniform weights, you should use another bucket algorithm.
-
-	#. **list**: List buckets aggregate their content as linked lists. Based on
-	   the :abbr:`RUSH (Replication Under Scalable Hashing)` :sub:`P` algorithm,
-	   a list is a natural and intuitive choice for an **expanding cluster**:
-	   either an object is relocated to the newest device with some appropriate
-	   probability, or it remains on the older devices as before. The result is
-	   optimal data migration when items are added to the bucket. Items removed
-	   from the middle or tail of the list, however, can result in a signiﬁcant
-	   amount of unnecessary movement, making list buckets most suitable for
-	   circumstances in which they **never (or very rarely) shrink**.
-
-	#. **tree**: Tree buckets use a binary search tree. They are more efficient
-	   than list buckets when a bucket contains a larger set of items. Based on
-	   the :abbr:`RUSH (Replication Under Scalable Hashing)` :sub:`R` algorithm,
-	   tree buckets reduce the placement time to O(log :sub:`n`), making them
-	   suitable for managing much larger sets of devices or nested buckets.
-
-	#. **straw**: List and Tree buckets use a divide and conquer strategy
-	   in a way that either gives certain items precedence (e.g., those
-	   at the beginning of a list) or obviates the need to consider entire
-	   subtrees of items at all. That improves the performance of the replica
-	   placement process, but can also introduce suboptimal reorganization
-	   behavior when the contents of a bucket change due an addition, removal,
-	   or re-weighting of an item. The straw bucket type allows all items to
-	   fairly “compete” against each other for replica placement through a
-	   process analogous to a draw of straws.
-
-        #. **straw2**: Straw2 buckets improve Straw to correctly avoid any data
-           movement between items when neighbor weights change.
-
-           For example the weight of item A including adding it anew or removing
-           it completely, there will be data movement only to or from item A.
+   Ceph supports five bucket types. Each bucket type provides a balance between
+   performance and reorganization efficiency, and each is different from the
+   others.  If you are unsure of which bucket type to use, use the ``straw2``
+   bucket. For a more technical discussion of bucket types than is offered
+   here, see **Section 3.4** of `CRUSH - Controlled, Scalable, Decentralized
+   Placement of Replicated Data`_. 
+   
+   The bucket types are as follows:
+
+    #. **uniform**: Uniform buckets aggregate devices that have **exactly** 
+       the same weight. For example, when hardware is commissioned or
+       decommissioned, it is often done in sets of machines that have exactly
+       the same physical configuration (this can be the case, for example,
+       after bulk purchases).  When storage devices have exactly the same
```

Comment:
@zdover23 - A puzzling introduction of double spaces.

Reply:
Fixed.